client/core/trade.go

```diff
@@ -527,6 +527,50 @@ func (t *trackedTrade) counterPartyConfirms(match *matchTracker) (have, needed u
 	return
 }
 
+// deleteStaleCancelOrder checks if this trade has an associated cancel order,
+// and deletes the cancel order if it (i.e. the cancel order) stays at Epoch
+// status for 2 or more epochs after the close of the cancel order's epoch.
+// Deleting the stale cancel order from this trade makes it possible for the
+// client to re-attempt cancelling the order.
+//
+// NOTE:
+// Stale cancel orders would be Executed if their preimage was sent or Revoked
+// if their preimages was not sent. We cannot currently tell whether the cancel
+// order's preimage was revealed, so assume that the cancel order is Executed
+// but unmatched. Consider adding a order.PreimageRevealed field to ensure that
+// the correct final status is set for the cancel order.
```

Comment:
"Consider adding a order.PreimageRevealed field to ensure that the correct final status is set for the cancel order"
...or allow the server to check status of cancel orders.